src/m365/spo/commands/site/site-recyclebinitem-move.ts

```diff
@@ -0,0 +1,201 @@
+    return commands.SITE_RECYCLEBINITEM_MOVE;
+  }
+
+  public get description(): string {
+    return 'Moves items from the first-stage recycle bin to the second-stage recycle bin';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initOptionSets();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        ids: typeof args.options.ids !== 'undefined',
+        all: !!args.options.all,
+        confirm: !!args.options.confirm
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --siteUrl <siteUrl>'
+      },
+      {
+        option: '-i, --ids [ids]'
+      },
+      {
+        option: '--all'
+      },
+      {
+        option: '--confirm'
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      { options: ['ids', 'all'] }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.siteUrl);
+        if (isValidSharePointUrl !== true) {
+          return isValidSharePointUrl;
+        }
+
+        if (args.options.ids && !validation.isValidGuidArray(args.options.ids.split(','))) {
+          return 'ids contains invalid GUID';
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Moving items from the first-stage recycle bin to the second-stage recycle bin at ${args.options.siteUrl}...`);
+    }
+
+    if (args.options.confirm) {
+      await this.moveRecycleBinItem(args, logger);
+    }
+    else {
+      const result = await Cli.prompt<{ continue: boolean }>({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: 'Are you sure you want to move the items to the second-stage recycle bin?'
+      });
+
+      if (result.continue) {
+        await this.moveRecycleBinItem(args, logger);
+      }
+    }
+  }
+
+  private async moveRecycleBinItem(args: CommandArgs, logger: Logger): Promise<void> {
+    try {
+      if (args.options.all) {
+        if (this.verbose) {
+          logger.logToStderr('Moving all items to the second-stage recycle bin');
+        }
+        const requestOptions: CliRequestOptions = {
+          url: `${args.options.siteUrl}/_api/web/recycleBin/MoveAllToSecondStage`,
+          headers: {
+            'accept': 'application/json;odata=nometadata'
+          },
+          responseType: 'json'
+        };
+        const response = await request.post<{ 'odata.null': boolean }>(requestOptions);
+        if (!response['odata.null']) {
+          throw 'Something went wrong when moving the selected items to the second-stage recycle bin';
+        }
```

Comment:
Seems like I made a little mistake here. When the API fails, we are getting a bad request code. So I guess this error check is not really needed anymore.